pkg/registry/cluster/storage/proxy.go

```diff
@@ -109,6 +109,9 @@ func newProxyHandler(location *url.URL, transport http.RoundTripper, impersonate
 
 		req.Header.Set("Authorization", fmt.Sprintf("bearer %s", impersonateToken))
 
+		// retain RawQuery in location because upgrading the request will use it
```

Comment:
I think that we can post this review here to explain the reason why we need to retain `RawQuery`:
https://github.com/karmada-io/karmada/issues/1618#issuecomment-1103793290

Reply:
Modified.